File: lang_specs/python/options/class/max.py
from abc import ABC, abstractmethod

class Abstract{classname}(ABC):
    @abstractmethod
    def greet(self) -> None:
        pass

class {classname}(Abstract{classname}):
    def __init__(self, message: str = "Hello, World!") -> None:
        self._message = message
    
    @property
    def message(self) -> str:
        return self._message
    
    @message.setter
    def message(self, value: str) -> None:
        self._message = value
    
    def greet(self) -> None:
        print(self.message)

if __name__ == "__main__":
    hw = {classname}()
    hw.greet()

File: lang_specs/rust/options/class/max.rs
pub struct {classname} {
    message: String,
}

impl Default for {classname} {
    fn default() -> Self {
        Self::new()
    }
}

impl {classname} {
    pub fn new() -> Self {
        Self {
            message: String::from("Hello, World!"),
        }
    }
    
    pub fn with_message(message: impl Into<String>) -> Self {
        Self {
            message: message.into(),
        }
    }
    
    pub fn greet(&self) {
        println!("{}", self.message);
    }
    
    pub fn message(&self) -> &str {
        &self.message
    }
}